apps/ledger-live-mobile/src/newArch/components/CopyButton/index.tsx

```diff
@@ -0,0 +1,59 @@
+import Clipboard from "@react-native-clipboard/clipboard";
+import React, { memo, useCallback, useMemo } from "react";
+import Animated, { useSharedValue, useAnimatedStyle, withTiming } from "react-native-reanimated";
+import styled from "styled-components/native";
+import { Check, Copy } from "@ledgerhq/icons-ui/native";
+import { Button } from "@ledgerhq/native-ui";
+import { ButtonProps } from "@ledgerhq/native-ui/components/cta/Button";
+
+export default memo(CopyButton);
+
+type Props = Omit<ButtonProps, "Icon" | "isNewIcon" | "onPress"> & {
+  text: string;
+  transitionDuration?: number;
+};
+
+function CopyButton({ text, ...props }: Props) {
+  const transition = useSharedValue(0);
+  const handleCopy = useCallback(() => {
+    Clipboard.setString(text);
+
+    transition.value = withTiming(1, { duration: 200 });
+    setTimeout(() => (transition.value = withTiming(0, { duration: 200 })), 1200);
+  }, [text, transition]);
+
+  const copyIconAnimation = useAnimatedStyle(() => ({
+    opacity: 1 - transition.value,
+  }));
+
+  const checkIconAnimation = useAnimatedStyle(() => ({
+    opacity: transition.value,
+  }));
+
+  const icon = useMemo(
+    () => (
+      <IconContainer>
+        <Animated.View style={copyIconAnimation}>
+          <Copy />
+        </Animated.View>
+        <Animated.View style={[checkIconAnimation, checkIconStyle]}>
+          <Check />
```

Comment:
Maybe use Icons.Check instead of importing the icon directly

Reply:
I imported Icons instead of each icon individualy. I'll keep that in mind in the future.
For the color green it felt a bit too flashy compare to the rest of the UI IMO.